project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/services/MinimizationPsiManagerService.kt

```diff
@@ -66,38 +83,54 @@ class MinimizationPsiManagerService {
      *
      * @param context The context for the minimization process containing the current project and relevant properties.
      * @param compressOverridden If set to true, then all overridden elements will be compressed to one element
+     * @param withFunctionParameters
      * @return A list of deletable PSI items found in the Kotlin files of the project.
      */
     suspend fun findDeletablePsiItems(
         context: IJDDContext,
         compressOverridden: Boolean = true,
+        withFunctionParameters: Boolean = false,
     ): List<PsiStubDDItem> = smartReadAction(context.indexProject) {
         if (compressOverridden) {
             findDeletablePsiItemsCompressed(context)
         } else {
-            findDeletablePsiItemsWithoutCompression(context).map { it.second }
+            findDeletablePsiItemsWithoutCompression(context, withFunctionParameters).map { it.item }
         }
     }
 
     @RequiresReadLock
-    private fun findDeletablePsiItemsWithoutCompression(context: IJDDContext) =
-        findPsiInKotlinFiles(context, PsiStubDDItem.DELETABLE_PSI_JAVA_CLASSES)
+    private fun findDeletablePsiItemsWithoutCompression(
+        context: IJDDContext,
+        withFunctionParameters: Boolean,
+    ): List<IntermediatePsiItemInfo> {
+        val coreElements = findPsiInKotlinFiles(context, PsiStubDDItem.DELETABLE_PSI_JAVA_CLASSES)
             .mapNotNull { psiElement ->
-                PsiUtils.buildDeletablePsiItem(context, psiElement).getOrNull()?.let { psiElement to it }
+                PsiUtils.buildDeletablePsiItem(context, psiElement).getOrNull()
+                    ?.let { IntermediatePsiItemInfo(psiElement, it) }
             }
+        logger.debug { "Got ${coreElements.size} core elements" }
+        val functionParameters = getFunctionsProperties(
```

Comment:
It's better to place a check of `withFunctionParameters` before calling `getFunctionsProperties`